templates/_components/_image.twig

```diff
@@ -3,13 +3,154 @@
+	2. Auto-crops images if needed/specified
+	3. Auto-generates art-directed responsive images using the `<picture>` and `<source>` tags
+	4. Auto-generates WebP versions (if supported)
+	5. Lazyload images using lazysizes
+	6. Allows images to be specified as CSS class background images
+	7. Allows use of different source images for different breakpoints
+	8. And finally, renders the image
+
+	How to use:
+  -----------
+	Include the `_image` component with the following parameters where you want to display the image.
+
+	Configuration parameters:
+	@param   asset                {AssetFileModel}          [Required] Image asset to be rendered
+	@param   breakpoint_assets    {array|AssetFileModel}    Array of image assets to be rendered for different breakpoints
+	@param   preset               {string}                  Pre-defined image transformation configuration name
+	@param   class                {string}                  Class attribute for the <img> tag
+	@param   alt                  {string}                  Alt attribute for the <img> tag
+	@param   lazyload             {bool}                    Whether to lazyload images (default: true)
+	@param   background           {string}                  Class name for background image
+	@param   webp                 {bool}                    Whether to generate WebP versions (default: true)
+
+	Example of simple usage:
+  --------------
+	```
+  <figure>
+    {% include '_components/_image' with { params: {
+      asset: image,
+      class: 'profile-pic',
+      preset: 'avatar',
+    } } %}
+	</figure>
+  ```
+
+	Example of usage with breakpoint asset:
+	------------------------------------
+	```
+	<figure>
+	{% include '_components/_image' with { params: {
+    asset: image,
+    breakpoint_assets: {
+     '--lg-plus': imageLarge,
+		 '--md-plus': imageMedium,
+		 '--sm-plus': imageSmall,
+	  },
+  } } %}
+	</figure>
+	```
+	
+	Set `presets`(a set of image transformation configurations) variable values in `_image` component.
+	Define the desired preset values in the image component as shown in the examples below.
+
+	Preset options:
+	---------------
+	@option   media          {string}      Any valid media query
+	@option   breakpoints    {array}       Array of transform values for different view breakpoints
+	@option   sizes          {array}       Array of size based transform values
+	@option   transform      {array}       Array of aspect and position related transform values
+	@option   guide          {string}      Any valid media query with size
+	
+	Preset examples:
+	---------------
+
+	1) Simple size-based transforms (can be width, or height — everything that the Imager plugin supports)
+		```
+		{% set presets = {
+			  avatar: {
+			    sizes: [
+			      { width: 20 },
+			      { width: 30 },
+			      { width: 40 },
+				   ],
+				 },	
+	  } %}
+		```
+
+		HTML Output:
+		`<img src=”small.jpg” srcset=”small.jpg 20w, medium.jpg 30w, large.jpg 40w”>`
```

Comment:
I think the file in the `src` attribute will probably be the largest generated transform, not the smallest. Please verify with this exact configuration and update the output.

Reply:
This will impact the `src` values in all the examples below.

Reply:
Tested and updated all `<img>` examples.